### PR TITLE
Extracted expression-language

### PR DESCRIPTION
**Pull request by wachterjohannes**
This PR extract the expression-language from tab-component. Used in https://github.com/sulu/sulu/pull/2967

### CODE REVIEW

File: husky_services/expression-language.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    var instance,
+        getInstance = function() {
+            if (instance == null) {
+                instance = new ExpressionLanguage();
+            }
+
+            return instance;
+        };
+
+    function ExpressionLanguage() {
```
Comment (danrot) [inline husky_services/expression-language.js]: Is `ExpressionLanguage` really a matching term? We are only evaluating based on javascript object and not introducing another kind of language.

File: husky_services/expression-language.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    'use strict';
+
+    var instance,
+        getInstance = function() {
+            if (instance == null) {
+                instance = new ExpressionLanguage();
+            }
+
+            return instance;
```
Comment (danrot) [inline husky_services/expression-language.js]: A require module is already a singleton, so there is no need to do something like this.

File: husky_services/logger.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+define(function() {
+
+    'use strict';
+
+    var instance = null,
+        getInstance = function() {
+            if (instance == null) {
+                instance = new Logger();
+            }
+            return instance;
```
Comment (danrot) [inline husky_services/logger.js]: Same as above, require modules are already singletons.

File: husky_services/logger.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        };
+
+    function Logger() {
+    }
+
+    Logger.prototype = window.Husky.logger;
```
Comment (danrot) [inline husky_services/logger.js]: How is the `logger` injected in Husky? Is there no better way than retrieving it from the `window` variable?

Comment (wachterjohannes) [inline husky_services/logger.js]: the logger will be initialized by aura and i have no really way to get it over require. we have done the same with mediator also.

**Comment by wachterjohannes** [comment]
@danrot fixed

File: .grunt/husky.json [hunk 5]
```diff
@@ -32,6 +32,8 @@
             "services/husky/mediator": "husky_services/mediator",
             "services/husky/translator": "husky_services/translator",
             "services/husky/url-validator": "husky_services/url-validator",
+            "services/husky/logger": "husky_services/logger",
+            "services/husky/expression": "husky_services/expression-language",
```
Comment (danrot) [inline .grunt/husky.json]: `expression`

**Comment by wachterjohannes** [comment]
fixed

**Pull request merged**